File: 1098-largest-unique-number/1098-largest-unique-number.py
from collections import defaultdict;

class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        dct = defaultdict(int)
        for n in nums:
            dct[n] += 1
        max_val = -1
        for key, val in dct.items():
            if val == 1:
                max_val = max(max_val, key)
        return max_val

File: 1098-largest-unique-number/1098-largest-unique-number.rs
use std::collections::HashMap;
impl Solution {
    pub fn largest_unique_number(nums: Vec<i32>) -> i32 {
        let mut freq : HashMap<i32, i32> = HashMap::new();

        for &n in nums.iter(){
            *freq.entry(n).or_insert(0)+= 1;
        }

        let mut largest = -1;
        for (&key, &val) in freq.iter(){
            if val == 1{
                largest = largest.max(key);
            }
        }
        largest
    }
}